Task1

grigorii-ch opened the pull request:
added:
  - depends postgresql and liquibase to root pom.xml
  - liquibase-maven-plugin to db module pom.xml
  - init.sql
  - database_example_changelog.xml (changeSet)

xgrisha committed:
Task1
 added:
  - depends postgresql and liquibase to root pom.xml
  - liquibase-maven-plugin to db module pom.xml
  - init.sql
  - database_example_changelog.xml (changeSet)

tlt-java-courses commented on blank-project-db/pom.xml:
```diff
@@ -28,6 +28,43 @@
+        </dependency>
     </dependencies>
 
+    <properties>
```
В мавене проперти идут до зависимостей

grigorii-ch replied:
Поправил

tlt-java-courses commented on blank-project-db/pom.xml:
```diff
@@ -28,6 +28,43 @@
     </dependencies>
 
+    <properties>
+        <db.host>localhost</db.host>
```
Это проверти для liquibase а не db. Давай переименуем

grigorii-ch replied:
Поправил

tlt-java-courses commented on blank-project-db/src/main/resources/db/database_example_changelog.xml:
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Давай назовем данный класс не с `example` а названием твоего проекта

grigorii-ch replied:
Инзменил на database_blankproject_changelog.xml

tlt-java-courses commented on blank-project-db/src/main/resources/db/database_example_changelog.xml:
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<databaseChangeLog
```
Давай сделаем этот ченжлог комулятивным, т.е. он будет сбором с инклюдами других ченжлогов

grigorii-ch replied:
Разложил по разным ченжлогам

tlt-java-courses commented on blank-project-db/src/main/resources/db/database_example_changelog.xml:
```diff
@@ -0,0 +1,58 @@
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd">
+    <changeSet id="1" author="Grigoriy">
```
1. Давай id назовем как-то более говоряще, хотя бы дату с описанием происходящего.
2. Давай запишем в авторстве Имя_Фамилию 
3. Если у тебя будет ошибка в создании одной из таблиц, откатятся все изменения в рамках скрипта миграции. Давай разделим 1 большой на много маленьких, хотя бы 1 скрипт миграции на одну таблицу

grigorii-ch replied:
Изменил id
добавил  Имя_Фамилию
раскидал по разным ченжлогам +  добавил preConditions на проверку существования таблицы с таким именем

tlt-java-courses commented on pom.xml:
```diff
@@ -117,7 +131,7 @@
                                         <limit>
                                             <counter>LINE</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.7</minimum>
```
Зачем?

grigorii-ch replied:
Вернул, не помню зачем менял :)

tlt-java-courses commented on blank-project-db/src/main/resources/init.sql:
```diff
@@ -0,0 +1,5 @@
+drop schema if exists chuprynin;
```
Не придирка, и сам не понимаю, но многие ратуют что ключевае слова должны быть в UPPER_CASE

grigorii-ch replied:
Перевел в UPPER_CASE

xgrisha committed:
Task1 refactoring
- change to upper case init.sql file
- breakdown to files changelogs
- chage to lower case table and field names in changelog
- some change in pom.xml

tlt-java-courses commented on blank-project-db/src/main/resources/db/database.changelog.v1.customer.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<databaseChangeLog
```
Зачем столько ченжлогов? Давай сделаем папку с названием условно v1 и туда сложим ещё один ченжлог, который будет иметь в названии версию v1.0 и содержать в себе перечень ченжсетов

grigorii-ch replied:
Не уверен что правильно понял,
Добавил папку v1 туда вынес подкорневой ченджлог и в него списал остальные, ну и + имена им покороче сделал
